### PR TITLE
fix(Script/ToC5): Remove hardcoded text(s) and some minor adjustments.

### LINKED ISSUE #4162: Core/Creature: hardcoded gossips in trial_of_the_champion
##### CURRENT BEHAVIOUR:

https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp#L11-L14

Need to convert to DB so locales can make use of it

AC HASH/COMMIT: 79aa560
OPERATING SYSTEM: w10

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/95701912-core-creature-hardcoded-gossips-in-trial_of_the_champion?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #4163: Core/Creature: hardcoded emote in boss_argent_challange
##### CURRENT BEHAVIOUR:

https://github.com/azerothcore/azerothcore-wotlk/blob/79aa5606547d898542557a01a2150f76e477e9e9/src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp#L75
Need to convert to DB so locales can make use of it

AC HASH/COMMIT: 79aa560
OPERATING SYSTEM: w10

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/95701911-core-creature-hardcoded-emote-in-boss_argent_challange?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by avarishd**
Delete cache before testing, test with non english client. (preferably Chinese client(only one with populated _locale)
Check the Chinese order, I was dual wielding google translate there.
## Changes Proposed:
-  Remove hardcoded text & fix DB duplicates (duplicated creature_text without broadcasts
- Example of 1 npc, all of them are in this state
c0a8a6cf594b2e3e2f0cc305c73b8773

## Issues Addressed:

- Closes #4162
- Closes #4163

## Tests Performed:

- tested in-game

## How to Test the Changes:

1. run ToC5 on Horde / Alliance

## Known Issues and TODO List:

- [ ] A bunch of things that needs sniffing, unrelated to the PR, minor things (such as RP) left some comments to whoever touches this next
- [x] Chinese translation GroupIDs need to be changed for most npcs (already exists in db, just needed re-linking)

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/instance_trial_of_the_champion.cpp [hunk 1]
```diff
@@ -25,6 +25,17 @@
 const Position SpawnPosition = {746.67f, 684.08f, 412.5f, 4.65f};
 #define CLEANUP_CHECK_INTERVAL  5000
 
+/*
+    Missing dialog/RP (already populated in DB) && spawns (can use ToC25 locations?) for:
+
```
Comment (Maelthyrr) [inline src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/instance_trial_of_the_champion.cpp:30]: Could you add "@ todo: " ? It's for doxygen documentation.

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_black_knight.cpp [hunk 2]
```diff
@@ -251,8 +251,8 @@ class boss_black_knight : public CreatureScript
                 case EVENT_ANNOUNCER_SAY_ZOMBIE:
                     if( pInstance && !summons.empty() )
                         if( Creature* ghoul = pInstance->instance->GetCreature(*summons.begin()) )
-                            ghoul->Yell("[Zombie] .... . Brains ....", LANG_UNIVERSAL);
-
+                            if (urand(0, 1))
+                                ghoul->Yell("[Zombie] .... . Brains ....", LANG_UNIVERSAL); // @todo: Multiple variations + not always happening, from video sources, needs sniff to transition from DB.
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_black_knight.cpp:255]: use 3 /

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/instance_trial_of_the_champion.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/instance_trial_of_the_champion.cpp:30]: Doxygen won't recognize this. You need
```
/**
 * @todo: Something
 *
 *
*/

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.h [hunk 3]
```diff
@@ -141,11 +141,28 @@ enum eNpcs
...
-    NPC_JAEREN                                      = 35004,
-    NPC_ARELAS                                      = 35005,
+    NPC_JAEREN                                      = 35004, // Horde
+    NPC_ARELAS                                      = 35005, // Alliance
     NPC_RISEN_JAEREN                                = 35545,
     NPC_RISEN_ARELAS                                = 35564,
-    NPC_TIRION                                      = 33628,
+    NPC_TIRION                                      = 33628, // Possibly wrong npc here, 34996 had already populated creature_text (from ToC25) that matches. Needs a sniff to confirm.
+
+    // @todo: Argent Raid Spectator - FX - Missing spawns for Toc5 (map 650), ToC25 probably matches positions, needs a sniff to confirm.
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.h:150]: 3 /

Comment (avarishd) [reply src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.h:150]: Ah, didn't know. Thanks!

**Comment by Maelthyrr** [comment]
Hi, thanks for the PR. Everythings works fine :)

**Review by Kitzunu — APPROVED** [review]

**Comment by pklloveyou** [comment]
Tested

**Pull request merged**